frame/ethereum/src/lib.rs

```diff
@@ -904,12 +910,18 @@ impl<T: Config> Pallet<T> {
 			transaction_data.into(),
 			weight_limit,
 			proof_size_base_cost,
-		)
-		.validate_in_block_for(&who)
-		.and_then(|v| v.with_chain_id())
-		.and_then(|v| v.with_base_fee())
-		.and_then(|v| v.with_balance_for(&who))
-		.map_err(|e| TransactionValidityError::Invalid(e.0))?;
+		);
+		check_transaction
+			.validate_in_block_for(&who)
+			.and_then(|v| v.with_chain_id())
+			.and_then(|v| v.with_base_fee())
+			.and_then(|v| v.with_balance_for(&who))
+			.map_err(|e| TransactionValidityError::Invalid(e.0))?;
+
+		use pallet_evm::OnChargeEVMTransaction;
+		let max_withdraw = check_transaction.max_withdraw_amount().map_err(|e| e.0)?;
+		<T as pallet_evm::Config>::OnChargeTransaction::can_withdraw(&origin, max_withdraw)
```

Comment:
What additional validation does this offer compared to the `with_balance_for` mentioned above?

Reply:
I'm rechecking this change we have at Moonbeam's `frontier` fork to see if the issue was fixed with https://github.com/paritytech/polkadot-sdk/pull/2292. Basically it meant to make a consistent check for the withdraw amount to prevent a corner case where the TX would be included in the pool even if it did not have enough balance to pay the fees.